src/proto/streams/prioritize.rs

```diff
@@ -694,13 +686,20 @@ impl Prioritize {
                              )
                         ),
                         None => {
-                            let reason = stream.state.get_scheduled_reset()
-                                .expect("must be scheduled to reset");
-
-                            stream.state.set_reset(reason);
-
-                            let frame = frame::Reset::new(stream.id, reason);
-                            Frame::Reset(frame)
+                            if let Some(reason) = stream.state.get_scheduled_reset() {
+                                stream.state.set_reset(reason);
+
+                                let frame = frame::Reset::new(stream.id, reason);
+                                Frame::Reset(frame)
+                            } else {
+                                // If the stream receives a RESET from the peer, it may have
+                                // had data buffered to be sent, but all the frames are cleared
+                                // in clear_queue(). Instead of doing O(N) traversal through queue
+                                // to remove, lets just ignore the stream here.
+                                trace!("removing dangling stream from pending_send");
+                                counts.transition_after(stream, is_counted, is_pending_reset);
```

Comment:
Could you explain why this needs to be called here? I don't *think* this was called before (and I could believe it was another bug).

Reply:
I agree that it was not called before. I think that it should be, otherwise we could leak streams - it's possible that the `pending_send` queue here was holding onto the last reference to the stream.

Reply:
I think it is highly likely that this change is correct, but i'd like to see if we can figure out a failing test case.